packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/recent-workspace.tsx

```diff
@@ -63,46 +66,65 @@ const RecentWorkspace = observer(({ workspace, index }: TRecentWorkspace) => {
         setSelectedStrategyId(workspace.id);
     };
 
+    const handleInit = () => {
+        setPreviewedStrategyId(workspace?.id);
+        if (active_tab === 0) {
+            previewRecentStrategy(workspace.id);
+        }
+    };
+
+    const handlePreviewList = () => {
+        setPreviewedStrategyId(workspace.id);
+        if (is_mobile) {
+            setPreviewOnDialog(true);
+            const dashboard_tab_dom_element = document.getElementsByClassName('tab__dashboard')?.[0];
+            waitForDomElement('#load-strategy__blockly-container', dashboard_tab_dom_element).then(() => {
+                previewRecentStrategy(workspace.id);
+            });
+        }
+    };
+
+    const handleEdit = async () => {
+        await loadFileFromRecent();
+        setActiveTab(DBOT_TABS.BOT_BUILDER);
+    };
+
+    const handleSave = () => {
+        updateBotName(workspace?.name);
+        toggleSaveModal();
+    };
+
     const viewRecentStrategy = async (type: string) => {
         setSelectedStrategyId(workspace.id);
+
         switch (type) {
-            case STRATEGY.INIT: {
-                setPreviewedStrategyId(workspace?.id);
-                // Fires for desktop
```

Comment:
@farabi-deriv can we keep these 2 comments? The init only fires for desktop and the preview list only fires for mobile after clicking the preview button on mobile. It may help in the future for debugging purposes.

Reply:
Sure. ✅